go/mysql/flavor.go

```diff
@@ -187,25 +194,29 @@ func GetFlavor(serverVersion string, flavorFunc func() flavor) (f flavor, capabl
 		f = flavorFunc()
 	case strings.HasPrefix(serverVersion, mariaDBReplicationHackPrefix):
 		canonicalVersion = serverVersion[len(mariaDBReplicationHackPrefix):]
-		f = mariadbFlavor101{}
+		f = mariadbFlavor101{mariadbFlavor{serverVersion: canonicalVersion}}
```

Comment:
I think we should have test coverage for what the expected behavior is if there's a future use of the flavor as we did before, i.e. with no `serverVersion` specified like it was here.

Reply:
Added.